src/masternode-payments.cpp

```diff
@@ -675,57 +664,35 @@ bool CMasternodePayments::ProcessBlock(int nBlockHeight)
 {
     if(!fMasterNode) return false;
 
-    //reference node - hybrid mode
-
-    if(!IsReferenceNode(activeMasternode.vin)){
```

Comment:
Are you sure this code block should be completely removed? Maybe it should run 100% of a time now instead? Because otherwise every single masternode (not just top MNPAYMENTS_SIGNATURES_TOTAL as it's now) will calculate new winner and relay it. I'm pretty sure they will be almost immediately banned for doing so (5+ blocks).